build01/marvin-mk2.nix

```diff
@@ -0,0 +1,65 @@
+{ pkgs, lib, config, ... }:
+let
+  userLib = import ../users/lib.nix { inherit lib; };
+
+  sources = import ../nix/sources.nix;
+  marvin-mk2 = pkgs.python3.pkgs.buildPythonApplication rec {
+    pname = "marvin-mk2";
+    version = "rolling";
+    src = import sources.marvin-mk2 { };
+
+    propagatedBuildInputs = with pkgs.python3.pkgs; [
+      aiohttp
+      gidgethub
+    ];
+  };
+
+  marvin-mk2-runtimeenv = with pkgs; [
+    marvin-mk2
+  ];
+
+in
+{
+  users.users.marvin-mk2.packages = [
+  ];
+  users.groups.marvin-mk2 = { };
+  users.users.marvin-mk2 = {
+    useDefaultShell = true;
+    uid = userLib.mkUid "mmk2";
+    extraGroups = [ "marvin-mk2" ];
+  };
+  nix.trustedUsers = [
+    "marvin-mk2"
+  ];
+
+  systemd.services.marvin-mk2 = {
+    description = "marvin-mk2 service";
+    enable = true;
+    path = marvin-mk2-runtimeenv;
+    environment.XDG_CONFIG_HOME = "/var/lib/marvin-mk2";
+    environment.XDG_CACHE_HOME = "/var/cache/marvin-mk2";
+    # FIXME
+    environment.GH_KEY_FILE = "";
+    # FIXME
+    environment.GH_APP_ID_FILE = "";
```

Comment:
By the way you can use EnvironmentFile to pass secrets from a file.

Reply:
That's quite nice. Potentially it means that the secret could even be owned by root isn't it?
I had a quick look at the [EnvironmentFile documentation](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#EnvironmentFile=) but it doesn't mention what user it will use to read the file.

Reply:
Oh, nice! Thanks.

Reply:
Having it owned by root might be nice, but since its exposed in the environment anyways the security benefits are probably marginal.